auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonAdapterFactoryProcessor.java

```diff
@@ -135,15 +137,24 @@ private TypeSpec createTypeAdapterFactory(
         .returns(result)
         .addStatement("Class<$T> rawType = (Class<$T>) $N.getRawType()", t, t, type);
 
-    for (int i = 0, elementsSize = elements.size(); i < elementsSize; i++) {
-      Element element = elements.get(i);
+    List<Map.Entry<Element, Optional<ExecutableElement>>> properties = elements.stream()
+        .collect(Collectors.toMap(e -> e, this::getTypeAdapterMethod))
+        .entrySet()
+        .stream()
+        .filter(entry -> entry.getValue().isPresent())
```

Comment:
Java 9 destroys this terrible pattern with `flatMap(Optional::stream)`. I'm tempted to even say that you should replace it with `flatMap(entry -> entry.map(Stream::of).orElse(Stream.empty())` to avoid using `isPresent` and `get`.